File: backend/ShiftMate.API/Models/ShiftRoleRequirement.cs
namespace ShiftMate.API.Models;

public class ShiftRoleRequirement
{
    public int Id { get; set; }

    public int ShiftId { get; set; }

    public int RoleId { get; set; } 

    public int RequiredEmployees { get; set; }

    public Shift Shift { get; set; } = null!;

    public Role Role { get; set; } = null!;
}

File: frontend/src/types/shiftRoleRequirement.ts
export interface ShiftRoleRequirement {
    id: number
    shiftId: number
    roleId: number
    roleName: string
    requiredEmployees: number
}

export interface CreateShiftRoleRequirementRequest {
    shiftId: number
    roleId: number
    requiredEmployees: number
}